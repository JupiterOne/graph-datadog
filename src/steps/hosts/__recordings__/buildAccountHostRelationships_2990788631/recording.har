{
  "log": {
    "_recordingName": "buildAccountHostRelationships",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "6.0.5"
    },
    "entries": [
      {
        "_id": "733d8fbbeb40b63451c5c0938bddaa27",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "datadog-api-client-typescript/1.9.0 (node 14.21.2; os darwin; arch x64)"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "dd-api-key",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "dd-application-key",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.us5.datadoghq.com"
            }
          ],
          "headersSize": 357,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.us5.datadoghq.com/api/v1/hosts"
        },
        "response": {
          "bodySize": 3370,
          "content": {
            "mimeType": "application/json",
            "size": 3370,
            "text": "{\"host_list\":[{\"id\":11654684,\"tags_by_source\":{\"Datadog\":[\"host:spoulton-MBP.local\"]},\"aliases\":[\"spoulton-MBP.local\"],\"apps\":[\"agent\",\"ntp\"],\"sources\":[\"agent\"],\"name\":\"spoulton-MBP.local\",\"host_name\":\"spoulton-MBP.local\",\"up\":true,\"last_reported_time\":1675970245,\"is_muted\":false,\"mute_timeout\":null,\"meta\":{\"agent_checks\":[[\"io\",\"io\",\"io\",\"OK\",\"\",\"\"],[\"cpu\",\"cpu\",\"cpu\",\"OK\",\"\",\"\"],[\"ntp\",\"ntp\",\"ntp:3c427a42a70bbf8\",\"OK\",\"\",\"\"],[\"network\",\"network\",\"network:4b0649b7e11f0772\",\"OK\",\"\",\"\"],[\"load\",\"load\",\"load\",\"OK\",\"\",\"\"],[\"disk\",\"disk\",\"disk:67cc0574430a16ba\",\"OK\",\"\",\"\"],[\"uptime\",\"uptime\",\"uptime\",\"OK\",\"\",\"\"],[\"memory\",\"memory\",\"memory\",\"OK\",\"\",\"\"]],\"agent_version\":\"7.42.1\",\"machine\":\"amd64\",\"cpuCores\":8,\"fbsdV\":[null,null,null],\"nixV\":[null,null,null],\"pythonV\":\"3.8.16\",\"agent_flavor\":\"agent\",\"platform\":\"darwin\",\"logs_agent\":{\"auto_multi_line_detection_enabled\":false,\"transport\":\"\"},\"gohai\":\"{\\\"cpu\\\":{\\\"cpu_cores\\\":\\\"8\\\",\\\"cpu_logical_processors\\\":\\\"16\\\",\\\"family\\\":\\\"6\\\",\\\"mhz\\\":\\\"2300\\\",\\\"model\\\":\\\"158\\\",\\\"model_name\\\":\\\"Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz\\\",\\\"stepping\\\":\\\"13\\\",\\\"vendor_id\\\":\\\"GenuineIntel\\\"},\\\"filesystem\\\":[{\\\"kb_size\\\":\\\"976797816\\\",\\\"mounted_on\\\":\\\"/\\\",\\\"name\\\":\\\"/dev/disk1s1s1\\\"},{\\\"kb_size\\\":\\\"976797816\\\",\\\"mounted_on\\\":\\\"/System/Volumes/Preboot\\\",\\\"name\\\":\\\"/dev/disk1s3\\\"},{\\\"kb_size\\\":\\\"976797816\\\",\\\"mounted_on\\\":\\\"/System/Volumes/VM\\\",\\\"name\\\":\\\"/dev/disk1s5\\\"},{\\\"kb_size\\\":\\\"976797816\\\",\\\"mounted_on\\\":\\\"/System/Volumes/Update\\\",\\\"name\\\":\\\"/dev/disk1s6\\\"},{\\\"kb_size\\\":\\\"976797816\\\",\\\"mounted_on\\\":\\\"/System/Volumes/Data\\\",\\\"name\\\":\\\"/dev/disk1s2\\\"}],\\\"memory\\\":{\\\"swap_total\\\":\\\"11264.00M\\\",\\\"total\\\":\\\"17179869184\\\"},\\\"network\\\":{\\\"interfaces\\\":[{\\\"ipv4\\\":[],\\\"ipv6\\\":[\\\"fe80::aede:48ff:fe00:1122\\\"],\\\"ipv6-network\\\":\\\"fe80::/64\\\",\\\"macaddress\\\":\\\"ac:de:48:00:11:22\\\",\\\"name\\\":\\\"en7\\\"},{\\\"ipv4\\\":[\\\"192.168.0.101\\\"],\\\"ipv4-network\\\":\\\"192.168.0.0/24\\\",\\\"ipv6\\\":[\\\"fe80::1087:6c6e:ae2d:6e64\\\"],\\\"ipv6-network\\\":\\\"fe80::/64\\\",\\\"macaddress\\\":\\\"5c:52:30:9d:2e:7e\\\",\\\"name\\\":\\\"en0\\\"},{\\\"ipv4\\\":[],\\\"ipv6\\\":[\\\"fe80::64dd:61ff:fe61:b36a\\\"],\\\"ipv6-network\\\":\\\"fe80::/64\\\",\\\"macaddress\\\":\\\"66:dd:61:61:b3:6a\\\",\\\"name\\\":\\\"awdl0\\\"},{\\\"ipv4\\\":[],\\\"ipv6\\\":[],\\\"name\\\":\\\"en2\\\"},{\\\"ipv4\\\":[],\\\"ipv6\\\":[],\\\"name\\\":\\\"en1\\\"},{\\\"ipv4\\\":[],\\\"ipv6\\\":[],\\\"name\\\":\\\"en3\\\"},{\\\"ipv4\\\":[],\\\"ipv6\\\":[],\\\"name\\\":\\\"en4\\\"},{\\\"ipv4\\\":[],\\\"ipv6\\\":[],\\\"name\\\":\\\"bridge0\\\"},{\\\"ipv4\\\":[],\\\"ipv6\\\":[\\\"fe80::64dd:61ff:fe61:b36a\\\"],\\\"ipv6-network\\\":\\\"fe80::/64\\\",\\\"macaddress\\\":\\\"66:dd:61:61:b3:6a\\\",\\\"name\\\":\\\"llw0\\\"},{\\\"ipv4\\\":[],\\\"ipv6\\\":[\\\"fe80::c803:8de1:77c5:66c7\\\"],\\\"ipv6-network\\\":\\\"fe80::/64\\\",\\\"name\\\":\\\"utun0\\\"},{\\\"ipv4\\\":[],\\\"ipv6\\\":[\\\"fe80::5dce:f773:b48c:e1ff\\\"],\\\"ipv6-network\\\":\\\"fe80::/64\\\",\\\"name\\\":\\\"utun1\\\"},{\\\"ipv4\\\":[],\\\"ipv6\\\":[\\\"fe80::ce81:b1c:bd2c:69e\\\"],\\\"ipv6-network\\\":\\\"fe80::/64\\\",\\\"name\\\":\\\"utun2\\\"},{\\\"ipv4\\\":[\\\"192.168.0.114\\\"],\\\"ipv4-network\\\":\\\"192.168.0.0/24\\\",\\\"ipv6\\\":[\\\"fe80::14af:eaeb:9b56:a318\\\"],\\\"ipv6-network\\\":\\\"fe80::/64\\\",\\\"macaddress\\\":\\\"64:4b:f0:20:81:96\\\",\\\"name\\\":\\\"en5\\\"},{\\\"ipv4\\\":[\\\"192.168.243.62\\\"],\\\"ipv4-network\\\":\\\"192.168.243.0/24\\\",\\\"ipv6\\\":[],\\\"name\\\":\\\"utun3\\\"}],\\\"ipaddress\\\":\\\"192.168.0.101\\\",\\\"ipaddressv6\\\":\\\"fe80::aede:48ff:fe00:1122\\\",\\\"macaddress\\\":\\\"5c:52:30:9d:2e:7e\\\"},\\\"platform\\\":{\\\"GOOARCH\\\":\\\"amd64\\\",\\\"GOOS\\\":\\\"darwin\\\",\\\"goV\\\":\\\"1.18.9\\\",\\\"hostname\\\":\\\"spoulton-MBP.local\\\",\\\"kernel_name\\\":\\\"Darwin\\\",\\\"kernel_release\\\":\\\"22.2.0\\\",\\\"kernel_version\\\":\\\"Darwin Kernel Version 22.2.0: Fri Nov 11 02:08:47 PST 2022; root:xnu-8792.61.2~4/RELEASE_X86_64\\\",\\\"machine\\\":\\\"x86_64\\\",\\\"os\\\":\\\"Darwin\\\",\\\"processor\\\":\\\"i386\\\"}}\",\"socket-hostname\":\"spoulton-MBP.local\",\"timezones\":[\"MST\"],\"macV\":[\"13.1\",[\"\",\"\",\"\"],\"amd64\"],\"winV\":[null,null,null],\"install_method\":{\"tool\":\"install_script_mac\",\"installer_version\":\"install_script_mac-1.1.0\",\"tool_version\":\"install_script_mac\"},\"socket-fqdn\":\"localhost\",\"processor\":\"Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz\",\"network\":null,\"host_id\":11654684},\"metrics\":{\"cpu\":6.2091756,\"iowait\":0,\"load\":0.24104817}},{\"id\":11909091,\"tags_by_source\":{\"Datadog\":[\"host:docker-desktop\"]},\"aliases\":[\"docker-desktop\"],\"apps\":[\"ntp\",\"agent\",\"container\",\"docker\"],\"sources\":[\"agent\"],\"name\":\"docker-desktop\",\"host_name\":\"docker-desktop\",\"up\":true,\"last_reported_time\":1675969918,\"is_muted\":false,\"mute_timeout\":null,\"meta\":{\"macV\":[\"\",\"\",\"\"],\"agent_version\":\"7.42.1\",\"network\":null,\"pythonV\":\"3.8.16\",\"machine\":\"amd64\",\"socket-fqdn\":\"dcef9da35b3a\",\"winV\":[\"\",\"\",\"\"],\"cpuCores\":1,\"gohai\":\"{\\\"cpu\\\":{\\\"cache_size\\\":\\\"12288 KB\\\",\\\"cpu_cores\\\":\\\"6\\\",\\\"cpu_logical_processors\\\":\\\"6\\\",\\\"family\\\":\\\"6\\\",\\\"mhz\\\":\\\"2600.000\\\",\\\"model\\\":\\\"158\\\",\\\"model_name\\\":\\\"Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz\\\",\\\"stepping\\\":\\\"10\\\",\\\"vendor_id\\\":\\\"GenuineIntel\\\"},\\\"filesystem\\\":[{\\\"kb_size\\\":\\\"61255492\\\",\\\"mounted_on\\\":\\\"/\\\",\\\"name\\\":\\\"overlay\\\"},{\\\"kb_size\\\":\\\"65536\\\",\\\"mounted_on\\\":\\\"/dev\\\",\\\"name\\\":\\\"tmpfs\\\"},{\\\"kb_size\\\":\\\"1016480\\\",\\\"mounted_on\\\":\\\"/sys/fs/cgroup\\\",\\\"name\\\":\\\"tmpfs\\\"},{\\\"kb_size\\\":\\\"65536\\\",\\\"mounted_on\\\":\\\"/dev/shm\\\",\\\"name\\\":\\\"shm\\\"},{\\\"kb_size\\\":\\\"61255492\\\",\\\"mounted_on\\\":\\\"/etc/hosts\\\",\\\"name\\\":\\\"/dev/vda1\\\"},{\\\"kb_size\\\":\\\"203300\\\",\\\"mounted_on\\\":\\\"/var/run/docker.sock\\\",\\\"name\\\":\\\"tmpfs\\\"},{\\\"kb_size\\\":\\\"10240\\\",\\\"mounted_on\\\":\\\"/host/sys/fs/cgroup\\\",\\\"name\\\":\\\"cgroup_root\\\"},{\\\"kb_size\\\":\\\"1016480\\\",\\\"mounted_on\\\":\\\"/proc/acpi\\\",\\\"name\\\":\\\"tmpfs\\\"},{\\\"kb_size\\\":\\\"1016480\\\",\\\"mounted_on\\\":\\\"/sys/firmware\\\",\\\"name\\\":\\\"tmpfs\\\"}],\\\"memory\\\":{\\\"swap_total\\\":\\\"1048572kB\\\",\\\"total\\\":\\\"2032964kB\\\"},\\\"network\\\":null,\\\"platform\\\":{\\\"GOOARCH\\\":\\\"amd64\\\",\\\"GOOS\\\":\\\"linux\\\",\\\"goV\\\":\\\"1.18.9\\\",\\\"hardware_platform\\\":\\\"x86_64\\\",\\\"hostname\\\":\\\"78f875a26a88\\\",\\\"kernel_name\\\":\\\"Linux\\\",\\\"kernel_release\\\":\\\"5.10.47-linuxkit\\\",\\\"kernel_version\\\":\\\"#1 SMP Sat Jul 3 21:51:47 UTC 2021\\\",\\\"machine\\\":\\\"x86_64\\\",\\\"os\\\":\\\"GNU/Linux\\\",\\\"processor\\\":\\\"x86_64\\\"}}\",\"container_meta\":{\"docker_swarm\":\"inactive\",\"docker_version\":\"20.10.10\"},\"agent_flavor\":\"agent\",\"processor\":\"Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz\",\"socket-hostname\":\"dcef9da35b3a\",\"install_method\":{\"tool\":\"docker\",\"installer_version\":\"docker\",\"tool_version\":\"docker\"},\"platform\":\"linux\",\"agent_checks\":[[\"network\",\"network\",\"network:4b0649b7e11f0772\",\"OK\",\"\",\"\"],[\"file_handle\",\"file_handle\",\"file_handle\",\"OK\",\"\",\"\"],[\"uptime\",\"uptime\",\"uptime\",\"OK\",\"\",\"\"],[\"docker\",\"docker\",\"docker\",\"OK\",\"\",\"\"],[\"ntp\",\"ntp\",\"ntp:3c427a42a70bbf8\",\"OK\",\"\",\"\"],[\"memory\",\"memory\",\"memory\",\"OK\",\"\",\"\"],[\"load\",\"load\",\"load\",\"OK\",\"\",\"\"],[\"container\",\"container\",\"container\",\"OK\",\"\",\"\"],[\"cpu\",\"cpu\",\"cpu\",\"OK\",\"\",\"\"],[\"io\",\"io\",\"io\",\"OK\",\"\",\"\"],[\"disk\",\"disk\",\"disk:67cc0574430a16ba\",\"OK\",\"\",\"\"]],\"fbsdV\":[\"\",\"\",\"\"],\"timezones\":[\"UTC\"],\"logs_agent\":{\"auto_multi_line_detection_enabled\":false,\"transport\":\"\"},\"nixV\":[\"ubuntu\",\"22.04\",\"\"],\"host_id\":11909091},\"metrics\":{\"cpu\":1.5792421,\"iowait\":0.0034869383,\"load\":0.020140741}}],\"total_returned\":2,\"total_matching\":2,\"exact_total_matching\":true}\n"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Thu, 09 Feb 2023 19:30:17 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "vary",
              "value": "Accept-Encoding"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=15724800;"
            },
            {
              "name": "content-security-policy",
              "value": "[REDACTED]"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "via",
              "value": "1.1 google"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 599,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2023-02-09T19:30:16.377Z",
        "time": 601,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 601
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
