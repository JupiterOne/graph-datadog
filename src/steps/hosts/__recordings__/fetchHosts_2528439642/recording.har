{
  "log": {
    "_recordingName": "fetchHosts",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "6.0.5"
    },
    "entries": [
      {
        "_id": "167cbcfd7a0cbcd710110676134158ff",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "dd-api-key",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "dd-application-key",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.us5.datadoghq.com"
            }
          ],
          "headersSize": 359,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "count",
              "value": "1000"
            },
            {
              "name": "start",
              "value": "0"
            }
          ],
          "url": "https://api.us5.datadoghq.com/api/v1/hosts?count=1000&start=0"
        },
        "response": {
          "bodySize": 2130,
          "content": {
            "mimeType": "application/json",
            "size": 2130,
            "text": "{\"host_list\":[{\"id\":11654684,\"tags_by_source\":{\"Datadog\":[\"host:spoulton-MBP.local\"]},\"aliases\":[\"spoulton-MBP.local\"],\"apps\":[\"agent\",\"ntp\"],\"sources\":[\"agent\"],\"name\":\"spoulton-MBP.local\",\"host_name\":\"spoulton-MBP.local\",\"up\":true,\"last_reported_time\":1675893200,\"is_muted\":false,\"mute_timeout\":null,\"meta\":{\"macV\":[\"13.1\",[\"\",\"\",\"\"],\"amd64\"],\"socket-hostname\":\"spoulton-MBP.local\",\"nixV\":[null,null,null],\"network\":null,\"processor\":\"Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz\",\"agent_checks\":[[\"io\",\"io\",\"io\",\"OK\",\"\",\"\"],[\"cpu\",\"cpu\",\"cpu\",\"OK\",\"\",\"\"],[\"ntp\",\"ntp\",\"ntp:3c427a42a70bbf8\",\"OK\",\"\",\"\"],[\"network\",\"network\",\"network:4b0649b7e11f0772\",\"OK\",\"\",\"\"],[\"load\",\"load\",\"load\",\"OK\",\"\",\"\"],[\"disk\",\"disk\",\"disk:67cc0574430a16ba\",\"OK\",\"\",\"\"],[\"uptime\",\"uptime\",\"uptime\",\"OK\",\"\",\"\"],[\"memory\",\"memory\",\"memory\",\"OK\",\"\",\"\"]],\"install_method\":{\"tool\":\"install_script_mac\",\"installer_version\":\"install_script_mac-1.1.0\",\"tool_version\":\"install_script_mac\"},\"agent_flavor\":\"agent\",\"machine\":\"amd64\",\"gohai\":\"{\\\"cpu\\\":{\\\"cpu_cores\\\":\\\"8\\\",\\\"cpu_logical_processors\\\":\\\"16\\\",\\\"family\\\":\\\"6\\\",\\\"mhz\\\":\\\"2300\\\",\\\"model\\\":\\\"158\\\",\\\"model_name\\\":\\\"Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz\\\",\\\"stepping\\\":\\\"13\\\",\\\"vendor_id\\\":\\\"GenuineIntel\\\"},\\\"filesystem\\\":[{\\\"kb_size\\\":\\\"976797816\\\",\\\"mounted_on\\\":\\\"/\\\",\\\"name\\\":\\\"/dev/disk1s1s1\\\"},{\\\"kb_size\\\":\\\"976797816\\\",\\\"mounted_on\\\":\\\"/System/Volumes/Preboot\\\",\\\"name\\\":\\\"/dev/disk1s3\\\"},{\\\"kb_size\\\":\\\"976797816\\\",\\\"mounted_on\\\":\\\"/System/Volumes/VM\\\",\\\"name\\\":\\\"/dev/disk1s5\\\"},{\\\"kb_size\\\":\\\"976797816\\\",\\\"mounted_on\\\":\\\"/System/Volumes/Update\\\",\\\"name\\\":\\\"/dev/disk1s6\\\"},{\\\"kb_size\\\":\\\"976797816\\\",\\\"mounted_on\\\":\\\"/System/Volumes/Data\\\",\\\"name\\\":\\\"/dev/disk1s2\\\"}],\\\"memory\\\":{\\\"swap_total\\\":\\\"11264.00M\\\",\\\"total\\\":\\\"17179869184\\\"},\\\"network\\\":{\\\"interfaces\\\":[{\\\"ipv4\\\":[],\\\"ipv6\\\":[\\\"fe80::aede:48ff:fe00:1122\\\"],\\\"ipv6-network\\\":\\\"fe80::/64\\\",\\\"macaddress\\\":\\\"ac:de:48:00:11:22\\\",\\\"name\\\":\\\"en7\\\"},{\\\"ipv4\\\":[\\\"192.168.0.101\\\"],\\\"ipv4-network\\\":\\\"192.168.0.0/24\\\",\\\"ipv6\\\":[\\\"fe80::1087:6c6e:ae2d:6e64\\\"],\\\"ipv6-network\\\":\\\"fe80::/64\\\",\\\"macaddress\\\":\\\"5c:52:30:9d:2e:7e\\\",\\\"name\\\":\\\"en0\\\"},{\\\"ipv4\\\":[],\\\"ipv6\\\":[\\\"fe80::fc6c:a2ff:fecd:ea99\\\"],\\\"ipv6-network\\\":\\\"fe80::/64\\\",\\\"macaddress\\\":\\\"fe:6c:a2:cd:ea:99\\\",\\\"name\\\":\\\"awdl0\\\"},{\\\"ipv4\\\":[],\\\"ipv6\\\":[],\\\"name\\\":\\\"en2\\\"},{\\\"ipv4\\\":[],\\\"ipv6\\\":[],\\\"name\\\":\\\"en1\\\"},{\\\"ipv4\\\":[],\\\"ipv6\\\":[],\\\"name\\\":\\\"en3\\\"},{\\\"ipv4\\\":[],\\\"ipv6\\\":[],\\\"name\\\":\\\"en4\\\"},{\\\"ipv4\\\":[],\\\"ipv6\\\":[],\\\"name\\\":\\\"bridge0\\\"},{\\\"ipv4\\\":[],\\\"ipv6\\\":[\\\"fe80::fc6c:a2ff:fecd:ea99\\\"],\\\"ipv6-network\\\":\\\"fe80::/64\\\",\\\"macaddress\\\":\\\"fe:6c:a2:cd:ea:99\\\",\\\"name\\\":\\\"llw0\\\"},{\\\"ipv4\\\":[],\\\"ipv6\\\":[\\\"fe80::c803:8de1:77c5:66c7\\\"],\\\"ipv6-network\\\":\\\"fe80::/64\\\",\\\"name\\\":\\\"utun0\\\"},{\\\"ipv4\\\":[],\\\"ipv6\\\":[\\\"fe80::5dce:f773:b48c:e1ff\\\"],\\\"ipv6-network\\\":\\\"fe80::/64\\\",\\\"name\\\":\\\"utun1\\\"},{\\\"ipv4\\\":[],\\\"ipv6\\\":[\\\"fe80::ce81:b1c:bd2c:69e\\\"],\\\"ipv6-network\\\":\\\"fe80::/64\\\",\\\"name\\\":\\\"utun2\\\"},{\\\"ipv4\\\":[\\\"192.168.0.114\\\"],\\\"ipv4-network\\\":\\\"192.168.0.0/24\\\",\\\"ipv6\\\":[\\\"fe80::14af:eaeb:9b56:a318\\\"],\\\"ipv6-network\\\":\\\"fe80::/64\\\",\\\"macaddress\\\":\\\"64:4b:f0:20:81:96\\\",\\\"name\\\":\\\"en5\\\"},{\\\"ipv4\\\":[],\\\"ipv6\\\":[],\\\"name\\\":\\\"en8\\\"}],\\\"ipaddress\\\":\\\"192.168.0.101\\\",\\\"ipaddressv6\\\":\\\"fe80::aede:48ff:fe00:1122\\\",\\\"macaddress\\\":\\\"5c:52:30:9d:2e:7e\\\"},\\\"platform\\\":{\\\"GOOARCH\\\":\\\"amd64\\\",\\\"GOOS\\\":\\\"darwin\\\",\\\"goV\\\":\\\"1.18.9\\\",\\\"hostname\\\":\\\"spoulton-MBP.local\\\",\\\"kernel_name\\\":\\\"Darwin\\\",\\\"kernel_release\\\":\\\"22.2.0\\\",\\\"kernel_version\\\":\\\"Darwin Kernel Version 22.2.0: Fri Nov 11 02:08:47 PST 2022; root:xnu-8792.61.2~4/RELEASE_X86_64\\\",\\\"machine\\\":\\\"x86_64\\\",\\\"os\\\":\\\"Darwin\\\",\\\"processor\\\":\\\"i386\\\"}}\",\"agent_version\":\"7.42.1\",\"timezones\":[\"MST\"],\"fbsdV\":[null,null,null],\"logs_agent\":{\"auto_multi_line_detection_enabled\":false,\"transport\":\"\"},\"pythonV\":\"3.8.16\",\"cpuCores\":8,\"winV\":[null,null,null],\"platform\":\"darwin\",\"socket-fqdn\":\"localhost\",\"host_id\":11654684},\"metrics\":{\"cpu\":12.230392,\"iowait\":0,\"load\":0.43045816}}],\"total_returned\":1,\"total_matching\":1,\"exact_total_matching\":true}\n"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Wed, 08 Feb 2023 22:17:41 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "vary",
              "value": "Accept-Encoding"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=15724800;"
            },
            {
              "name": "content-security-policy",
              "value": "[REDACTED]"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "via",
              "value": "1.1 google"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 599,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2023-02-08T22:17:41.304Z",
        "time": 234,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 234
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
