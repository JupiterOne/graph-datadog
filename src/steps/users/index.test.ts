import {
  createMockStepExecutionContext,
  Recording,
} from '@jupiterone/integration-sdk-testing';
import { fetchUsers } from '.';
import { integrationConfig } from '../../../test/config';
import { setupZendeskRecording } from '../../../test/recording';
import { IntegrationConfig } from '../../config';
import { Entities } from '../constants';

describe('#fetchUsers', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupZendeskRecording({
      directory: __dirname,
      name: 'fetchUsers',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchUsers(context);

    const users = context.jobState.collectedEntities.filter((e) =>
      e._type.includes(Entities.USER._type),
    );

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(users.length).toBeGreaterThan(0);
    expect(users).toMatchGraphObjectSchema({
      _class: Entities.USER._class,
      schema: {
        additionalProperties: false,
        required: [],
        properties: {
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          _type: { const: Entities.USER._type },
          _class: { const: Entities.USER._class },
          _key: { type: 'string' },
          username: { type: 'string' },
          email: { type: 'string' },
          id: { type: 'string' },
          name: { type: 'string' },
          verified: { type: 'boolean' },
          disabled: { type: 'boolean' },
          icon: { type: 'string' },
          displayName: { type: 'string' },
          accessRole: { type: 'string' },
        },
      },
    });
  });
});
