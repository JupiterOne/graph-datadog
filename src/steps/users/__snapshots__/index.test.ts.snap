// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchUsers should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "datadog_user:development@jupiterone.dev",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": User {
            "accessRole": "adm",
            "disabled": false,
            "email": "development@jupiterone.dev",
            "handle": "development@jupiterone.dev",
            "icon": "https://secure.gravatar.com/avatar/6db8032ecd87512265a13d4bc9e6483f?s=48&d=retro",
            "name": "JupiterOne Development",
            "verified": true,
          },
        },
      ],
      "_type": "datadog_user",
      "accessRole": "adm",
      "createdOn": undefined,
      "disabled": false,
      "displayName": "JupiterOne Development",
      "email": "development@jupiterone.dev",
      "icon": "https://secure.gravatar.com/avatar/6db8032ecd87512265a13d4bc9e6483f?s=48&d=retro",
      "id": "development@jupiterone.dev",
      "name": "JupiterOne Development",
      "username": "development@jupiterone.dev",
      "verified": true,
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "datadog_user:viragsf@gmail.com",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": User {
            "accessRole": "adm",
            "disabled": false,
            "email": "viragsf@gmail.com",
            "handle": "viragsf@gmail.com",
            "icon": "https://secure.gravatar.com/avatar/edd1469180957cee25d7753ef4a84a13?s=48&d=retro",
            "name": null,
            "verified": true,
          },
        },
      ],
      "_type": "datadog_user",
      "accessRole": "adm",
      "createdOn": undefined,
      "disabled": false,
      "displayName": "",
      "email": "viragsf@gmail.com",
      "icon": "https://secure.gravatar.com/avatar/edd1469180957cee25d7753ef4a84a13?s=48&d=retro",
      "id": "viragsf@gmail.com",
      "name": "",
      "username": "viragsf@gmail.com",
      "verified": true,
    },
  ],
  "collectedRelationships": Array [],
  "encounteredTypes": Array [
    "datadog_user",
  ],
  "numCollectedEntities": 2,
  "numCollectedRelationships": 0,
}
`;
